### ISSUE TITLE
Training questions

### ISSUE THREAD

**Issue by nom** [yisol/IDM-VTON#44]
Hey, great work! Quick question on training. I was wondering how you're fitting two SDXL UNets (garment UNet and tryon UNet) on a single A800 with batch size 24/4=6 (assuming 4xA800 in total). Are you doing any optimizations to bring memory down, like precomputing embeddings / features or 8bit adam? Also curious about training speed if you can share. Thanks!

**Comment by yisol** [comment]
Hello, we used gradient checkpointing and 8 bit adam for training and fit batch size 6 to single A100 GPU.
We didn't use precomputing latents and embeddings or gradient accumulation but you can use them for reducing memory cost.
Training time was around 1~2day on 4xA100 GPU for 63k iterations.

**Comment by nom** [comment]
Thanks @yisol. That's quite surprising to me. Are you perhaps not using EMA? 

Also if you could share a work-in-progress rough train script here, that'd be really helpful - just to get a better understanding of discrepancy, doesn't have to be a working script.

**Comment by ifeherva** [comment]
Did you use [noise_offset](https://www.crosslabs.org//blog/diffusion-with-offset-noise) or [snr_gamma](https://arxiv.org/abs/2303.09556) (=5) during training?

**Comment by cardosofelipe** [comment]
> Thanks @yisol. Are you perhaps not doing EMA?
> 
> Also if you could share a work-in-progress rough train script here, that'd be really helpful - just to get a better understanding of the differences with mine, doesn't have to be a working script.

@yisol It would really helpful indeed

**Comment by Anustup900** [comment]
Hey @nom I am trying to replicate the training, it would be great if you share a glimpse of your script or an idea also will work.

**Comment by awzhgw** [comment]
@nom  can you share finetune code for me ?

**Comment by jasonaidm** [comment]
@nom can you share training or finetune code for me ?

**Comment by thuc248997** [comment]
@nom can you share the fine-tune code?

**Comment by awzhgw** [comment]
@nom  can you share the fine-tune code?

**Comment by ttjygbtj22** [comment]
@nom can you share the fine-tune code?

**Comment by nftblackmagic** [comment]
I made an unofficial training code here. Still testing it. Please try if you like:
https://github.com/nftblackmagic/IDM-VTON-training

**Comment by shahzain110** [comment]
> I made an unofficial training code here. Still testing it. Please try if you like: https://github.com/nftblackmagic/IDM-VTON-training

brother this code is training the complete model? how can i only finetune the model?
